src/frontend/src/pages/GanttPage/GanttChartComponents/GanttChangeModals/GanttRequestChangeModal.tsx

```diff
@@ -0,0 +1,23 @@
+import { validateWBS } from 'shared';
+import { RequestEventChange } from '../../../../utils/gantt.utils';
+import { GanttProjectCreateModal } from './GanttProjectCreateModal';
+import { GanttTimeLineChangeModal } from './GanttTimeLineChangeModal';
+import { GanttWorkPackageCreateModal } from './GanttWorkPackageCreateModal';
+
+interface GanttRequestChangeModalProps {
+  change: RequestEventChange;
+  handleClose: () => void;
+  open: boolean;
+}
+
+export const GanttRequestChangeModal = ({ change, handleClose, open }: GanttRequestChangeModalProps) => {
+  if (change.createProject) {
+    return <GanttProjectCreateModal change={change} handleClose={handleClose} open={open} />;
+  }
+  try {
+    validateWBS(change.eventId);
+    return <GanttTimeLineChangeModal change={change} handleClose={handleClose} open={open} />;
+  } catch {
```

Comment:
don't love this, if there's a way to add the change type to RequestEventChange that'd be better, but the gantt chart code is pretty fucked, so if need be just make a ticket for it, and we can clean it up later